source/common/http/header_map_impl.h

```diff
@@ -340,6 +346,10 @@ class HeaderMapImpl : NonCopyable {
  */
 template <class Interface> class TypedHeaderMapImpl : public HeaderMapImpl, public Interface {
 public:
+  void setFormatter(StatefulHeaderKeyFormatterPtr&& formatter) {
+    formatter_ = std::move(formatter);
```

Comment:
Nice, was thinking we'd need to avoid touching HeaderMap at all, but this definitely works around the issue of having added memory implications for those not using the feature.